src/main/java/de/tum/in/www1/artemis/service/AssessmentService.java

```diff
@@ -232,34 +239,11 @@ public double calculateTotalScore(Double calculatedScore, Double maxScore) {
      */
     public Double calculateTotalScore(List<Feedback> assessments) {
         double totalScore = 0.0;
-
         var gradingInstructions = new HashMap<Long, Integer>(); // { instructionId: noOfEncounters }
+
         for (Feedback feedback : assessments) {
             if (feedback.getGradingInstruction() != null) {
-                if (gradingInstructions.get(feedback.getGradingInstruction().getId()) != null) {
-                    // We Encountered this grading instruction before
-                    var maxCount = feedback.getGradingInstruction().getUsageCount();
-                    var encounters = gradingInstructions.get(feedback.getGradingInstruction().getId());
-                    if (maxCount > 0) {
-                        if (encounters >= maxCount) {
-                            // the structured grading instruction was applied on assessment models more often that the usageCount limit allows so we don't sum the feedback credit
-                            gradingInstructions.put(feedback.getGradingInstruction().getId(), encounters + 1);
-                        }
-                        else {
-                            // the usageCount limit was not exceeded yet so we add the credit and increase the nrOfEncounters counter
-                            gradingInstructions.put(feedback.getGradingInstruction().getId(), encounters + 1);
-                            totalScore += feedback.getGradingInstruction().getCredits();
-                        }
-                    }
-                    else {
-                        totalScore += feedback.getCredits();
-                    }
-                }
-                else {
-                    // First time encountering the grading instruction
-                    gradingInstructions.put(feedback.getGradingInstruction().getId(), 1);
-                    totalScore += feedback.getCredits();
-                }
+                totalScore = gradingCriterionService.computeTotalScore(feedback, totalScore, gradingInstructions);
```

Comment:
I think I've seen this exact calculation also in the client `gradingCriterionService`. Could this be unified somehow? I guess it was always like this.

Reply:
See general comment.